packages/clerk-js/src/ui/components/UserProfile/EnterpriseAccountsSection.tsx

```diff
@@ -15,58 +31,110 @@ export const EnterpriseAccountsSection = () => {
-            >
-              <Image
-                elementDescriptor={[descriptors.providerIcon]}
-                elementId={descriptors.enterpriseButtonsProviderIcon.setId(account.provider)}
-                alt={providerName}
-                src={providerLogoUrl}
-                sx={theme => ({ width: theme.sizes.$4 })}
-              />
-              <Box sx={{ whiteSpace: 'nowrap', overflow: 'hidden' }}>
-                <Flex
-                  gap={2}
-                  center
-                >
-                  <Text
-                    truncate
-                    colorScheme='body'
-                  >
-                    {providerName}
-                  </Text>
-                  <Text
-                    truncate
-                    as='span'
-                    colorScheme='secondary'
-                  >
-                    {label ? `• ${label}` : ''}
-                  </Text>
-                  {error && (
-                    <Badge
-                      colorScheme='danger'
-                      localizationKey={localizationKeys('badge__requiresAction')}
-                    />
-                  )}
-                </Flex>
-              </Box>
-            </ProfileSection.Item>
-          );
-        })}
+        {activeEnterpriseAccounts.map(account => (
+          <EnterpriseAccount
+            key={account.id}
+            account={account}
+          />
+        ))}
       </ProfileSection.ItemList>
     </ProfileSection.Root>
   );
 };
+
+const EnterpriseAccount = ({ account }: { account: EnterpriseAccountResource }) => {
+  const label = account.emailAddress;
+  const providerName = getEnterpriseAccountProviderName(account);
+  const error = account.verification?.error?.longMessage;
+
+  return (
+    <ProfileSection.Item
+      id='enterpriseAccounts'
+      sx={t => ({
+        gap: t.space.$2,
+        justifyContent: 'start',
+      })}
+      key={account.id}
+    >
+      <EnterpriseAccountProviderIcon account={account} />
+      <Box sx={{ whiteSpace: 'nowrap', overflow: 'hidden' }}>
+        <Flex
+          gap={2}
+          center
+        >
+          <Text
+            truncate
+            colorScheme='body'
+          >
+            {providerName}
+          </Text>
+          <Text
+            truncate
+            as='span'
+            colorScheme='secondary'
+          >
+            {label ? `• ${label}` : ''}
+          </Text>
+          {error && (
+            <Badge
+              colorScheme='danger'
+              localizationKey={localizationKeys('badge__requiresAction')}
+            />
+          )}
+        </Flex>
+      </Box>
+    </ProfileSection.Item>
+  );
+};
+
+const EnterpriseAccountProviderIcon = ({ account }: { account: EnterpriseAccountResource }) => {
+  const { provider } = account;
+  const providerName = getEnterpriseAccountProviderName(account);
+
+  if (isOAuthCustomProvider(provider)) {
+    return (
+      <ProviderInitialIcon
+        id={provider}
```

Comment:
Maybe introduce a more generic `providerInitialIcon` descriptor?

```tsx
elementDescriptor={[descriptors.providerIcon, descriptors.providerInitialIcon]}
elementId={descriptors.enterpriseButtonsProviderIcon.setId(account.provider)}
```

Reply:
@alexcarpenter I've added a new description that it's generic for SSO connections, and not specific to enterprise or social.

Also in that commit, it includes a fix where we display the logo URL for custom OAuth providers that can be defined from the Clerk Dashboard.

Reply:
I think we need to preserve `socialButtonsProviderInitialIcon` descriptors for within social buttons since those could already be in use and would be a breaking change removing them.

Reply:
Gotcha, does it make sense to pass it as prop to `ProviderInitialIcon` then? 🤔

Reply:
Actually, I think it makes sense to provide as an array as you mentioned - I'll update here later.

Reply:
@alexcarpenter I've added a more generic description here and kept `socialButtonsProviderInitialIcon` [here](https://github.com/clerk/javascript/pull/4518/commits/3466922e99e279a1ee7c5578e0e2f1395d36a328)